### PR TITLE
Include CNUInstallDirs for correct install paths

### PR DESCRIPTION
**Pull request by ktrace**
Include CNUInstallDirs for correct install paths and some cleanups for trailing spaces.

### CODE REVIEW

File: distribution/posix/CMakeLists.txt [hunk 1]
```diff
@@ -5,53 +5,55 @@ SET(CPACK_SET_DESTDIR "ON")
...
+        DESTINATION "${CMAKE_INSTALL_DOCDIR}"
+        COMPONENT "application"
         PERMISSIONS OWNER_WRITE OWNER_READ GROUP_READ WORLD_READ
         ) 
 # MESSAGE("PROJECT_SOURCE_DIR ${PROJECT_SOURCE_DIR}")
 
 INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
-        DESTINATION "share/applications/" 
-        COMPONENT "application" 
+        DESTINATION "${CMAKE_INSTALL_DOCDIR}/applications/"
```
Comment (mgorny) [inline distribution/posix/CMakeLists.txt]: I think you meant DATAROOTDIR here.

File: desktop/ponyprog.desktop [hunk 2]
```diff
@@ -1,12 +1,12 @@
 [Desktop Entry]
-Version=3.0.0
+Version=3.0
```
Comment (mgorny) [inline desktop/ponyprog.desktop]: This should be `1.1` as its the version of desktop spec. See: https://standards.freedesktop.org/desktop-entry-spec/latest/ar01s06.html

File: desktop/ponyprog.desktop [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
 Name=Ponyprog
 Comment=EEPROM and microcontroller flasher
 GenericName=Ponyprog
```
Comment (mgorny) [inline desktop/ponyprog.desktop]: This 'generic name' doesn't make much sense as it's not generic. I think it'd make sense to drop it.

File: distribution/posix/CMakeLists.txt [hunk 4]
```diff
@@ -49,7 +49,7 @@ INSTALL(FILES "${PROJECT_SOURCE_DIR}/copyright"
 # MESSAGE("PROJECT_SOURCE_DIR ${PROJECT_SOURCE_DIR}")
 
 INSTALL(FILES "${PROJECT_SOURCE_DIR}/desktop/ponyprog.desktop"
-        DESTINATION "${CMAKE_INSTALL_DOCDIR}/applications/"
+        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/applications/"
```
Comment (Moiman) [inline distribution/posix/CMakeLists.txt]: This is in wrong commit

Comment (ktrace) [reply distribution/posix/CMakeLists.txt]: moved

**Review by lancos — COMMENTED** [review]
Why did you change version from 3.0.0 to 1.1?

**Comment by Moiman** [comment]
> Why did you change version from 3.0.0 to 1.1?

" Version of the Desktop Entry Specification that the desktop entry conforms with. Entries that confirm with this version of the specification should use 1.1. Note that the version field is not required to be present. "
https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html

**Review by eduard-x — APPROVED** [review]

**Comment by eduard-x** [comment]
CMake changes are correct. Thank you!

**Pull request merged**